tests/unit/backend/recipe/algorithm/test_FitMultiplePeaksAlgorithm.py

```diff
@@ -0,0 +1,51 @@
+import os
+import unittest.mock as mock
+
+with mock.patch.dict(
+    "sys.modules",
+    {
+        "snapred.backend.log": mock.Mock(),
+        "snapred.backend.log.logger": mock.Mock(),
+    },
+):
+    from mantid.simpleapi import LoadNexusProcessed
+    from snapred.backend.dao.calibration.Calibration import Calibration
+    from snapred.backend.dao.CrystallographicInfo import CrystallographicInfo
+    from snapred.backend.dao.FitMultiplePeaksIngredients import FitMultiplePeaksIngredients
+    from snapred.backend.recipe.algorithm.FitMultiplePeaksAlgorithm import (
+        FitMultiplePeaksAlgorithm,  # noqa: E402
+    )
+    from snapred.meta.Config import Resource
+
+    def test_init():
+        """Test ability to initialize purge overlapping peaks algo"""
+        instrumentState = Calibration.parse_raw(
+            Resource.read("/inputs/purge_peaks/input_parameters.json")
+        ).instrumentState
+        crystalInfo = CrystallographicInfo.parse_raw(Resource.read("/inputs/purge_peaks/input_crystalInfo.json"))
+        wsName = "testWS"
+        fitIngredients = FitMultiplePeaksIngredients(
+            InstrumentState=instrumentState, CrystalInfo=crystalInfo, InputWorkspace=wsName
+        )
+        fmpAlgo = FitMultiplePeaksAlgorithm()
+        fmpAlgo.initialize()
+        fmpAlgo.setProperty("FitMultiplePeaksIngredients", fitIngredients.json())
+        assert fmpAlgo.getProperty("FitMultiplePeaksIngredients").value == fitIngredients.json()
+
+    def test_execute():
+        inputFile = os.path.join(Resource._resourcesPath, "FitMultiplePeaksTestWS.nxs").replace("tests", "src/snapred")
+        LoadNexusProcessed(Filename=inputFile, OutputWorkspace="testWS")
+        instrumentState = Calibration.parse_raw(
+            Resource.read("/inputs/purge_peaks/input_parameters.json")
+        ).instrumentState
+        crystalInfo = CrystallographicInfo.parse_raw(Resource.read("/inputs/purge_peaks/input_crystalInfo.json"))
+        wsName = "testWS"
+        fitIngredients = FitMultiplePeaksIngredients(
+            InstrumentState=instrumentState, CrystalInfo=crystalInfo, InputWorkspace=wsName
+        )
+        fmpAlgo = FitMultiplePeaksAlgorithm()
+        fmpAlgo.initialize()
+        fmpAlgo.setProperty("FitMultiplePeaksIngredients", fitIngredients.json())
+        fmpAlgo.execute()
+        wsGroup = fmpAlgo.getProperty("OutputWorkspaceGroup").value
+        assert wsGroup == "fitPeaksWSGroup"
```

Comment:
Can we add better testing, such as ensuring proper values inside wsGroup?  At minimum, check what values *are* and assert those, so we can check future edits produce same output.

Reply:
I don't want to test the values of the resulting workspaces because then I am testing result of Mantid `FitPeaks` and there is already extensive testing done in mantid to ensure that algorithm is working. Also I get different resulting values depending on which version of Mantid I am using. I have updated the test to check that the list of workspaces created in the workspace group are the correct workspaces. That should be the same every time no matter the result of `FitPeaks`